ChefExtensionHandler/bin/chef-install.psm1

```diff
@@ -27,13 +27,33 @@ function Chef-GetExtensionRoot {
 }
 
 function Install-ChefClient {
-  trap [Exception] {echo $_.Exception.Message;exit 1}
-
-  iex (new-object net.webclient).downloadstring('https://omnitruck.chef.io/install.ps1');install
+  $retries = 3
+  $retrycount = 0
+  $completed = $false
+
+  while (-not $completed) {
+    echo "Downloading Chef Client ..."
+    Try {
+      iex (new-object net.webclient).downloadstring('https://omnitruck.chef.io/install.ps1');install
+      $completed = $true
+    }
+    Catch [System.Net.WebException] {
+      ## this catch is for the WebException raised during a WebClient request while downloading the chef-client package ##
+      if ($retrycount -ge $retries) {
+        echo "Chef Client Downloading failed after 3 retries."
+        $ErrorMessage = $_.Exception.Message
+        # log to CommandExecution log:
+        echo "Error running install: $ErrorMessage"
+        exit 1
+      } else {
+        echo "Chef Client package download failed. Retrying..."
+        $retrycount++
```

Comment:
We should sleep here, not just immediately retry. It's likely to fail again if we try immediately. I would give it 30s at most, to account for our total install time budget.